src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,286 @@
+                try {
+                    int taskNumber = Integer.parseInt(command.substring(7));
+                    TaskList.unmarkTask(taskNumber);
+                } catch (NumberFormatException e) {
+                    throw new TakoException("Wrong format! Unmark command should have the form 'unmark x'," +
+                                            " where x is an integer");
+                }
+            }
+            throw new TakoException("Wrong format! Unmark command should have the form 'unmark x'," +
+                                    " where x is an integer");
+        } catch (TakoException e) {
+            System.out.println(e.message());
+            Ui.promptInput();
+        }
+    }
+
+    /**
+     * Deletes the task according to the input from the list
+     *
+     * @param command
+     * @throws TakoException if the delete command is not in the right form
+     */
+    public static void parseDelete(String command) {
+        try {
+            if (command.length() > 6 && command.charAt(6) == ' ') {
+                try {
+                    int taskNumber = Integer.parseInt(command.substring(7));
+                    TaskList.deleteTask(taskNumber);
+                } catch (NumberFormatException e) {
+                    throw new TakoException("Wrong format! Delete command should have the form 'delete x'," +
+                                            " where x is an integer");
+                }
+            }
+            throw new TakoException("Wrong format! Delete command should have the form 'delete x'," +
+                                    " where x is an integer");
+        } catch (TakoException e) {
+            System.out.println(e.message());
+            Ui.promptInput();
+        }
+    }
+
+    /**
+     * Breaks down the command into its description
+     * Make the todo task according to the description
+     *
+     * @param command
+     * @throws TakoException if the todo command is not in the right form
+     */
+    public static void parseTodo(String command) {
+        try {
+            if (command.length() > 4 && command.charAt(4) == ' ' && !command.substring(5).isBlank()) {
+                String description = command.substring(5);
+                TaskList.addTask(new ToDo(description));
+            } else {
+                throw new TakoException("Wrong format! Todo command should have the form 'mark x'," +
+                                        " where x is not empty");
+            }
+        } catch (TakoException e) {
+            System.out.println(e.message());
+            Ui.promptInput();
+        }
+    }
+
+    /**
+     * Breaks down the command into its description and when it is due
+     * Make the deadline task according to the description and its due date
+     *
+     * @param command
+     * @throws TakoException if the deadline command is not in the right form
+     */
+    public static void parseDeadline(String command) {
```

Comment:
The deeply nested code may be difficult to read with really specific string checking - maybe some comments would be valuable?